### PR TITLE
Improve widget tests

### PR DESCRIPTION
**Pull request by d-loose**
~~WIP~~
Some test cleanup and additions

- cleanup dialog tests
- test device properties on `DevicePage`
- check whether the correct buttons are shown on `DevicePage`

### CODE REVIEW

**Comment by d-loose** [comment]
- test whether the action buttons do the right things
- test whether error and confirmation dialogs are shown in `error` and `needsReboot` states

File: test/device_page_test.dart [hunk 1]
```diff
@@ -105,6 +126,65 @@ void main() {
...
+      final notifier = mockNotifier();
+      await tester.pumpApp((_) => buildPage(model: model, notifier: notifier));
+      await tester.pumpAndSettle();
+
+      expect(find.text(tester.lang.rebootConfirm), findsOneWidget);
+      expect(find.text(tester.lang.reboot), findsOneWidget);
+      expect(find.text(tester.lang.cancel), findsOneWidget);
+
+      await tester.tap(find.text(tester.lang.cancel));
+      verify(model.state = DeviceState.idle).called(1);
```
Comment (jpnurmi) [inline test/device_page_test.dart:171]: could add `verifyNever(model.reboot());` here :)

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

File: test/device_page_test.dart (hunk 1 above)
Comment (d-loose) [reply test/device_page_test.dart:171]: :+1:

**Pull request merged**